Only named exports may use 'export type'.

With 0.1.9 the following code would be converted as follows:

JS:
```
// @flow
export type * from './types';
```

TS:
```
export * from './types';
```

But with 0.2.2 I get the following:
```
export type * from "./types";
```

which results in a TS error:

```
lib/index.d.ts:1:1 - error TS1383: Only named exports may use 'export type'.

1 export type * from "./types";
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/package.json
+++ b/package.json
@@ -37,7 +37,6 @@
     "jest": "^24.5.0",
     "jest-mock-console": "^0.4.2",
     "jest-mock-process": "^1.2.0",
-    "pretty-quick": "^1.10.0",
     "tmp": "^0.1.0",
     "typescript": "^3.4.5"
   },
@@ -50,7 +49,7 @@
   "husky": {
     "hooks": {
       "pre-push": "npm test",
-      "pre-commit": "pretty-quick --staged"
+      "pre-commit": "npm run pretty-quick && npm run pretty-tests"
     }
   },
   "jest": {

--- a/src/transform.js
+++ b/src/transform.js
@@ -708,16 +708,26 @@ const transform = {
   },
   ImportDeclaration: {
     exit(path) {
-      // TODO(#223): Handle "typeof" imports
-      if (path.node.importKind === "typeof") {
-        path.node.importKind = "value";
-      }
       stripSuffixFromImportSource(path);
+      if (
+        path.node.importKind === "typeof" &&
+        t.isImportDefaultSpecifier(path.node.specifiers[0])
+      ) {
+        path.replaceWith(
+          t.tsTypeAliasDeclaration(
+            path.node.specifiers[0].local,
+            undefined,
+            t.tsTypeQuery(
+              t.tsImportType(path.node.source, t.identifier("default"))
+            )
+          )
+        );
+      }
     },
   },
   ImportSpecifier: {
     exit(path) {
-      // TODO(#223): Handle "typeof" imports
+      // TODO(#223): Handle "typeof" imports.
       if (path.node.importKind === "typeof") {
         path.node.importKind = "value";
       }

--- a/src/transform.js
+++ b/src/transform.js
@@ -690,7 +690,16 @@ const transform = {
       path.replaceWith(t.tsExpressionWithTypeArguments(id, typeParameters));
     },
   },
-  ExportDeclaration: {
+  ExportAllDeclaration: {
+    exit(path) {
+      // TypeScript doesn't support `export type * from ...`
+      path.node.exportKind = "value";
+      if (path.node.source) {
+        stripSuffixFromImportSource(path);
+      }
+    },
+  },
+  ExportNamedDeclaration: {
     exit(path) {
       if (path.node.source) {
         stripSuffixFromImportSource(path);
@@ -700,12 +709,18 @@ const transform = {
   ImportDeclaration: {
     exit(path) {
       // TODO(#223): Handle "typeof" imports
+      if (path.node.importKind === "typeof") {
+        path.node.importKind = "value";
+      }
       stripSuffixFromImportSource(path);
     },
   },
   ImportSpecifier: {
     exit(path) {
       // TODO(#223): Handle "typeof" imports
+      if (path.node.importKind === "typeof") {
+        path.node.importKind = "value";
+      }
     },
   },
   DeclareVariable: {

--- a/test/fixtures/convert/imports/export_star_from/flow.js
+++ b/test/fixtures/convert/imports/export_star_from/flow.js
@@ -1,0 +1,1 @@
+export * from "./foo.js";

--- a/test/fixtures/convert/imports/export_star_from/ts.js
+++ b/test/fixtures/convert/imports/export_star_from/ts.js
@@ -1,0 +1,1 @@
+export * from "./foo";

--- a/test/fixtures/convert/imports/export_type/flow.js
+++ b/test/fixtures/convert/imports/export_type/flow.js
@@ -1,0 +1,1 @@
+export type A = string;

--- a/test/fixtures/convert/imports/export_type/ts.js
+++ b/test/fixtures/convert/imports/export_type/ts.js
@@ -1,0 +1,1 @@
+export type A = string;

--- a/test/fixtures/convert/imports/export_type_star_from/flow.js
+++ b/test/fixtures/convert/imports/export_type_star_from/flow.js
@@ -1,0 +1,1 @@
+export type * from "./types.js";

--- a/test/fixtures/convert/imports/export_type_star_from/ts.js
+++ b/test/fixtures/convert/imports/export_type_star_from/ts.js
@@ -1,0 +1,1 @@
+export * from "./types";

--- a/test/fixtures/convert/imports/import_typeof/flow.js
+++ b/test/fixtures/convert/imports/import_typeof/flow.js
@@ -1,0 +1,1 @@
+import typeof A from "./a.js";

--- a/test/fixtures/convert/imports/import_typeof/ts.js
+++ b/test/fixtures/convert/imports/import_typeof/ts.js
@@ -1,0 +1,1 @@
+import A from "./a";

--- a/test/fixtures/convert/imports/import_typeof/ts.js
+++ b/test/fixtures/convert/imports/import_typeof/ts.js
@@ -1,1 +1,1 @@
-import A from "./a";
+type A = typeof import("./a").default;

